homeassistant/components/upnp/config_flow.py

```diff
@@ -42,9 +42,9 @@ def _is_complete_discovery(discovery_info: ssdp.SsdpServiceInfo) -> bool:
     """Test if discovery is complete and usable."""
     return (
         ssdp.ATTR_UPNP_UDN in discovery_info.upnp
-        and discovery_info.ssdp_st
-        and discovery_info.ssdp_location
-        and discovery_info.ssdp_usn
+        and cast(bool, discovery_info.ssdp_st)
```

Comment:
This is circumventing the type checker a bit. Perhaps instead:
```suggestion
       return bool(
        ssdp.ATTR_UPNP_UDN in discovery_info.upnp
        and discovery_info.ssdp_st
        and discovery_info.ssdp_location
        and discovery_info.ssdp_usn
        and discovery_info.ssdp_st
        and discovery_info.ssdp_location
        and discovery_info.ssdp_usn
    )
```

Reply:
I have adjusted this